diff --git a/vl/generic.c b/vl/generic.c
index c6f84a9..d0f3db4 100644
--- a/vl/generic.c
+++ b/vl/generic.c
@@ -1512,15 +1512,7 @@ vl_thread_specific_state_delete (VlThreadState * self)
  * in different ways depending on the operating system.
  */
 
-#if (defined(VL_OS_LINUX) || defined(VL_OS_MACOSX)) && defined(VL_COMPILER_GNUC)
-static void vl_constructor () __attribute__ ((constructor)) ;
-static void vl_destructor () __attribute__ ((destructor))  ;
-#endif
-
-#if defined(VL_OS_WIN)
-static void vl_constructor () ;
-static void vl_destructor () ;
-
+#if defined(VL_OS_WIN) & ! defined(VL_STATIC)
 BOOL WINAPI DllMain(
     HINSTANCE hinstDLL,  // handle to DLL module
     DWORD fdwReason,     // reason for calling function
@@ -1563,7 +1555,7 @@ BOOL WINAPI DllMain(
 /* ---------------------------------------------------------------- */
 
 /** @internal @brief Initialize VLFeat state */
-static void
+void
 vl_constructor (void)
 {
   VlState * state ;
@@ -1637,7 +1629,7 @@ vl_constructor (void)
 }
 
 /** @internal @brief Destruct VLFeat */
-static void
+void
 vl_destructor ()
 {
   VlState * state ;
diff --git a/vl/generic.h b/vl/generic.h
index 4ef87f2..99c5e49 100644
--- a/vl/generic.h
+++ b/vl/generic.h
@@ -206,5 +206,14 @@ VL_EXPORT double vl_toc (void) ;
 VL_EXPORT double vl_get_cpu_time (void) ;
 /** @} */
 
+
+#if (defined(VL_OS_LINUX) || defined(VL_OS_MACOSX)) && defined(VL_COMPILER_GNUC)
+VL_EXPORT void vl_constructor() __attribute__ ((constructor));
+VL_EXPORT void vl_destructor() __attribute__ ((destructor));
+#else
+VL_EXPORT void vl_constructor();
+VL_EXPORT void vl_destructor();
+#endif
+
 /* VL_GENERIC_H */
 #endif
diff --git a/vl/host.h b/vl/host.h
index 98e20ca..d4ce0de 100644
--- a/vl/host.h
+++ b/vl/host.h
@@ -316,6 +316,9 @@ defined(__DOXYGEN__)
 #    define snprintf _snprintf
 #    define isnan _isnan
 #  endif
+# ifdef VL_STATIC
+#  define VL_EXPORT
+# else
 #  ifdef VL_BUILD_DLL
 #    ifdef __cplusplus
 #      define VL_EXPORT extern "C" __declspec(dllexport)
@@ -329,6 +332,7 @@ defined(__DOXYGEN__)
 #      define VL_EXPORT extern __declspec(dllimport)
 #    endif
 #  endif
+# endif
 #endif
 
 #if defined(VL_COMPILER_LCC) & ! defined(__DOXYGEN__)
diff --git a/vl/sift.c b/vl/sift.c
index 03963fe..3739f49 100644
--- a/vl/sift.c
+++ b/vl/sift.c
@@ -1443,8 +1443,9 @@ vl_sift_detect (VlSiftFilt * f)
  ** @remark The minimum octave size is 2x2xS.
  **/
 
-static void
-update_gradient (VlSiftFilt *f)
+#define update_gradient vl_sift_update_gradient
+void
+vl_sift_update_gradient (VlSiftFilt *f)
 {
   int       s_min = f->s_min ;
   int       s_max = f->s_max ;
diff --git a/vl/sift.h b/vl/sift.h
index 50e03f4..5d996d3 100644
--- a/vl/sift.h
+++ b/vl/sift.h
@@ -139,6 +139,9 @@ void  vl_sift_keypoint_init              (VlSiftFilt const *f,
                                           double sigma) ;
 /** @} */
 
+VL_EXPORT
+void vl_sift_update_gradient(VlSiftFilt* f);
+
 /** @name Retrieve data and parameters
  ** @{
  **/
