diff --git a/src/io/dataset_io.cpp b/src/io/dataset_io.cpp
--- a/src/io/dataset_io.cpp
+++ b/src/io/dataset_io.cpp
@@ -36,8 +36,10 @@
 #include <basalt/io/dataset_io.h>
 #include <basalt/io/dataset_io_euroc.h>
 #include <basalt/io/dataset_io_kitti.h>
-#include <basalt/io/dataset_io_rosbag.h>
 #include <basalt/io/dataset_io_uzh.h>
+#ifdef BASALT_USE_ROSBAG
+#include <basalt/io/dataset_io_rosbag.h>
+#endif
 
 namespace basalt {
 
@@ -46,8 +48,10 @@
   if (dataset_type == "euroc") {
     // return DatasetIoInterfacePtr();
     return DatasetIoInterfacePtr(new EurocIO(load_mocap_as_gt));
+#ifdef BASALT_USE_ROSBAG
   } else if (dataset_type == "bag") {
     return DatasetIoInterfacePtr(new RosbagIO);
+#endif
   } else if (dataset_type == "uzh") {
     return DatasetIoInterfacePtr(new UzhIO);
   } else if (dataset_type == "kitti") {
