--- a/acados/ocp_qp/ocp_qp_hpmpc.h
+++ b/acados/ocp_qp/ocp_qp_hpmpc.h
@@ -120,7 +120,7 @@
 //
 void ocp_qp_hpmpc_eval_forw_sens(void *config_, void *qp_in, void *qp_out, void *opts_, void *mem_, void *work_);
 //
-void ocp_qp_hpmpc_solver_get(void *config_, void *qp_in_, void *qp_out_, void *opts_, void *mem_, const char *field, int stage, void* value, int size1, int size2)
+void ocp_qp_hpmpc_solver_get(void *config_, void *qp_in_, void *qp_out_, void *opts_, void *mem_, const char *field, int stage, void* value, int size1, int size2);
 //
 
 //
