diff --git a/gds/CMakeLists.txt b/gds/CMakeLists.txt
index 019f9f891..d8248130a 100644
--- a/gds/CMakeLists.txt
+++ b/gds/CMakeLists.txt
@@ -24,6 +24,9 @@ if (NOT APPLE)
     set(CMAKE_INSTALL_RPATH $ORIGIN)
 endif ()
 
+# Find CUDA Toolkit for cudart and cufile.
+find_package(CUDAToolkit REQUIRED)
+
 ################################################################################
 # Add library: cufile_stub
 ################################################################################
@@ -52,52 +55,19 @@ target_compile_options(cufile_stub
         )
 
 ## Link libraries
-target_link_libraries(cufile_stub
+if (CUCIM_STATIC_GDS)
+    # Enabling CUCIM_STATIC_GDS statically links cuFile
+    target_link_libraries(cufile_stub
         PUBLIC
             ${CMAKE_DL_LIBS}
         PRIVATE
-            CUDA::cudart
-        )
-
-# Set GDS include path (cufile.h)
-if (DEFINED ENV{CONDA_BUILD} AND EXISTS $ENV{PREFIX}/include/cufile.h)
-    set(GDS_INCLUDE_PATH "$ENV{PREFIX}/include")
-elseif (DEFINED ENV{CONDA_PREFIX} AND EXISTS $ENV{CONDA_PREFIX}/include/cufile.h)
-    set(GDS_INCLUDE_PATH "$ENV{CONDA_PREFIX}/include")
-elseif (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/../temp/cuda/include/cufile.h)
-    set(GDS_INCLUDE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../temp/cuda/include)
-else ()
-    set(GDS_INCLUDE_PATH /usr/local/cuda/include)
-endif ()
-
-message("Set GDS_INCLUDE_PATH to '${GDS_INCLUDE_PATH}'.")
-
-# Enabling CUCIM_STATIC_GDS assumes that lib/libcufile_static.a and include/cufile.h is available
-# under ../temp/cuda folder.
-if (CUCIM_STATIC_GDS)
-    add_library(deps::gds_static STATIC IMPORTED GLOBAL)
-
-    if (DEFINED ENV{CONDA_BUILD})
-        set(GDS_STATIC_LIB_PATH "$ENV{PREFIX}/lib/libcufile_static.a")
-    elseif (DEFINED ENV{CONDA_PREFIX})
-        set(GDS_STATIC_LIB_PATH "$ENV{CONDA_PREFIX}/lib/libcufile_static.a")
-    elseif (EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/../temp/cuda/lib64/libcufile_static.a)
-        set(GDS_STATIC_LIB_PATH ${CMAKE_CURRENT_SOURCE_DIR}/../temp/cuda/lib64/libcufile_static.a)
-    else ()
-        set(GDS_STATIC_LIB_PATH /usr/local/cuda/lib64/libcufile_static.a)
-    endif ()
-
-    message("Set GDS_STATIC_LIB_PATH to '${GDS_STATIC_LIB_PATH}'.")
-
-    set_target_properties(deps::gds_static PROPERTIES
-        IMPORTED_LOCATION "${GDS_STATIC_LIB_PATH}"
-        INTERFACE_INCLUDE_DIRECTORIES "${GDS_INCLUDE_PATH}"
+            CUDA::cuFile_static
     )
+else()
+    # Use `dlopen` to load cuFile at runtime
     target_link_libraries(cufile_stub
         PUBLIC
             ${CMAKE_DL_LIBS}
-        PRIVATE
-            deps::gds_static
     )
 endif()
 
@@ -110,9 +80,8 @@ PUBLIC
 target_include_directories(cufile_stub
     PUBLIC
         $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
-        $<BUILD_INTERFACE:${GDS_INCLUDE_PATH}>
+        ${CUDAToolkit_INCLUDE_DIRS}
     PRIVATE
-        # Add path to cufile.h explicitly. ${TOP}/temp/cuda would be available by `./run copy_gds_files_`
         ${CMAKE_CURRENT_SOURCE_DIR}/../cpp/include # for including helper.h in cucim/dynlib
 )
 
